engine/src/exec-interface.cpp

```diff
@@ -4308,25 +4305,29 @@ bool MCInterfaceExecSortContainer(MCExecContext &ctxt, MCStringRef p_data, int p
         t_trailing_delim = true;
         t_item_count--;
 	}
-    
+	
+	MCAutoArray<MCStringRef> t_sorted;
 	// Sort the array
-	MCStringsExecSort(ctxt, p_direction, (Sort_type)p_form, *t_chunks, t_item_count, p_by, t_sorted, t_sorted_count);
-
+	MCStringsExecSort(ctxt, p_direction, (Sort_type)p_form,
+	                  *t_chunks, t_item_count, p_by,
+	                  t_sorted . PtrRef(), t_sorted . SizeRef());
```

Comment:
`MCStringsExecSort()` is a bit of an odd function, and it took me a while to work out what was going on here.

It might be worth adding a comment although `MCStringsExecSort()` allocates memory to hold the sorted array, it _doesn't_ retain each of the strings before putting them in the array -- so `t_sorted` has to free the sorted array only, and releasing the strings is taken care of by the destructor of `t_chunks` -- and that's why `t_sorted` can't be an `MCAutoStringRefArray`.